apps/aecore/lib/aecore/channel/channel_state_on_chain.ex

```diff
@@ -451,6 +455,40 @@ defmodule Aecore.Channel.ChannelStateOnChain do
     }
   end
 
+  @doc """
+  Validates the payload of ChannelSnapshotSoloTx.
+  """
+  @spec validate_snapshot(ChannelStateOnChain.t(), ChannelOffChainTx.t()) ::
+          :ok | {:error, binary()}
+  def validate_snapshot(
+        %ChannelStateOnChain{} = channel,
+        %ChannelOffChainTx{} = offchain_tx
+      ) do
+    if channel.sequence < offchain_tx.sequence do
+      ChannelOffChainTx.verify_signatures(offchain_tx, pubkeys(channel))
+    else
+      {:error,
+       "#{__MODULE__}: OffChainTx is too old, expected newer than #{channel.sequence}, got #{
+         offchain_tx.sequence
+       }"}
+    end
+  end
+
+  @doc """
+  Applies the provided payload of ChannelSnapshotSoloTx to the state of the channel. The contents should be validated by &validate_snapshot/2
+  """
+  @spec apply_snapshot(ChannelStateOnChain.t(), ChannelOffChainTx.t()) :: ChannelStateOnChain.t()
+  def apply_snapshot(%ChannelStateOnChain{} = channel, %ChannelOffChainTx{
+        sequence: sequence,
+        state_hash: state_hash
+      }) do
+    %ChannelStateOnChain{
+      channel
+      | sequence: sequence,
```

Comment:
According to line 168 of `aesc_channels.erl` you should set `sequence_solo` to 0.

```
snapshot_solo(Ch, PayloadTx) ->
    Round = aesc_offchain_tx:round(PayloadTx),
    StateHash = aesc_offchain_tx:state_hash(PayloadTx),
    Ch#channel{round              = Round,
               solo_round         = 0,
               state_hash         = StateHash}.
```